File: mission-to-mars/src/main/java/mars/Helicoptere.java
package mars;

public class Helicoptere extends Rover implements ModuleMartien {

	public Helicoptere(Direction direction, Position position) {
		super(direction, position);
	}

	@Override
	public void monter() {
		position = new Position(position.getX(), position.getY(), position.getZ() + 1);
	}

	@Override
	public void descendre() {
		position = new Position(position.getX(), position.getY(), position.getZ() - 1);
	}
}

File: python/mission-to-mars/src/mars/Helicoptere.py
from src.mars.Direction import Direction
from src.mars.ModuleMartien import ModuleMartien
from src.mars.Position import Position
from src.mars.Rover import Rover


class Helicoptere(Rover, ModuleMartien):

    def __init__(self, direction: Direction, position: Position):
        super().__init__(direction, position)

    def monter(self) -> None:
        self.position = Position(self.position.x, self.position.y, self.position.z + 1)

    def descendre(self) -> None:
        self.position = Position(self.position.x, self.position.y, self.position.z - 1)
